core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -288,6 +292,14 @@ static TableMetadata fromJson(FileIO io, InputFile file, JsonNode node) {
           schema, TableMetadata.INITIAL_SPEC_ID, node.get(PARTITION_SPEC)));
     }
 
+    int lastAssignedFieldId;
+    if (formatVersion > 1) {
+      lastAssignedFieldId = JsonUtil.getInt(LAST_ASSIGNED_FIELD_ID, node);
```

Comment:
Should this use a precondition to check that the last assigned partition ID is present? That would result in a better error message if it is missing, like the other cases: `last-assigned-partition-id must exist in format v2`